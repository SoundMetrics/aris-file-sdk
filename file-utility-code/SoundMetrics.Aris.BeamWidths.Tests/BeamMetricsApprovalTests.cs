using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoundMetrics.Aris.Headers;
using System.Text;

namespace SoundMetrics.Aris.BeamWidths.Tests
{
    [TestClass]
    [UseApprovalSubdirectory("approval-files"), UseReporter(typeof(DiffReporter))]
    public sealed class BeamMetricsApprovalTests
    {
        [TestMethod]
        public void BeamWidths_ARIS3000_128()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 128, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS3000_64()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 64, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.Telephoto));

        private static void RunTest(BeamConfig cfg)
        {
            var stringOutput = BuildStringOutput(cfg);
            Approvals.Verify(stringOutput);
        }

        private static string BuildStringOutput(BeamConfig cfg)
        {
            var beamInfos = Metrics.GetBeamInformation(cfg);
            var stringOutput = BuildOutput(beamInfos);
            return stringOutput;

            static string BuildOutput(BeamInfo[] beamInfos)
            {
                var buffer = new StringBuilder();
                buffer.AppendLine($"# of BeamInfo members: [{beamInfos.Length}]");

                foreach (var beamInfo in beamInfos)
                {
                    buffer.AppendLine(beamInfo.ToString());
                }

                return buffer.ToString();
            }
        }
    }
}
