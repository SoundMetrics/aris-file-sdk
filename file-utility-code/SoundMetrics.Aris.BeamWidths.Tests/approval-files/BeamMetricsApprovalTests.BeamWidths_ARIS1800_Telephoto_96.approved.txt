# of BeamInfo members: [96]
BeamInfo { BeamNumber = 0, Center = -7.419, Left = -7.493, Right = -7.345 }
BeamInfo { BeamNumber = 1, Center = -7.271, Left = -7.345, Right = -7.197 }
BeamInfo { BeamNumber = 2, Center = -7.123, Left = -7.197, Right = -7.048 }
BeamInfo { BeamNumber = 3, Center = -6.974, Left = -7.048, Right = -6.899 }
BeamInfo { BeamNumber = 4, Center = -6.824, Left = -6.899, Right = -6.749 }
BeamInfo { BeamNumber = 5, Center = -6.674, Left = -6.749, Right = -6.598 }
BeamInfo { BeamNumber = 6, Center = -6.523, Left = -6.598, Right = -6.447 }
BeamInfo { BeamNumber = 7, Center = -6.372, Left = -6.447, Right = -6.296 }
BeamInfo { BeamNumber = 8, Center = -6.22, Left = -6.296, Right = -6.144 }
BeamInfo { BeamNumber = 9, Center = -6.068, Left = -6.144, Right = -5.992 }
BeamInfo { BeamNumber = 10, Center = -5.915, Left = -5.992, Right = -5.839 }
BeamInfo { BeamNumber = 11, Center = -5.763, Left = -5.839, Right = -5.686 }
BeamInfo { BeamNumber = 12, Center = -5.609, Left = -5.686, Right = -5.533 }
BeamInfo { BeamNumber = 13, Center = -5.456, Left = -5.533, Right = -5.379 }
BeamInfo { BeamNumber = 14, Center = -5.302, Left = -5.379, Right = -5.225 }
BeamInfo { BeamNumber = 15, Center = -5.148, Left = -5.225, Right = -5.07 }
BeamInfo { BeamNumber = 16, Center = -4.993, Left = -5.07, Right = -4.915 }
BeamInfo { BeamNumber = 17, Center = -4.838, Left = -4.915, Right = -4.76 }
BeamInfo { BeamNumber = 18, Center = -4.682, Left = -4.76, Right = -4.604 }
BeamInfo { BeamNumber = 19, Center = -4.526, Left = -4.604, Right = -4.448 }
BeamInfo { BeamNumber = 20, Center = -4.37, Left = -4.448, Right = -4.291 }
BeamInfo { BeamNumber = 21, Center = -4.213, Left = -4.291, Right = -4.134 }
BeamInfo { BeamNumber = 22, Center = -4.055, Left = -4.134, Right = -3.976 }
BeamInfo { BeamNumber = 23, Center = -3.897, Left = -3.976, Right = -3.818 }
BeamInfo { BeamNumber = 24, Center = -3.739, Left = -3.818, Right = -3.66 }
BeamInfo { BeamNumber = 25, Center = -3.581, Left = -3.66, Right = -3.501 }
BeamInfo { BeamNumber = 26, Center = -3.422, Left = -3.501, Right = -3.343 }
BeamInfo { BeamNumber = 27, Center = -3.264, Left = -3.343, Right = -3.185 }
BeamInfo { BeamNumber = 28, Center = -3.105, Left = -3.185, Right = -3.026 }
BeamInfo { BeamNumber = 29, Center = -2.947, Left = -3.026, Right = -2.868 }
BeamInfo { BeamNumber = 30, Center = -2.788, Left = -2.868, Right = -2.709 }
BeamInfo { BeamNumber = 31, Center = -2.63, Left = -2.709, Right = -2.55 }
BeamInfo { BeamNumber = 32, Center = -2.471, Left = -2.55, Right = -2.392 }
BeamInfo { BeamNumber = 33, Center = -2.313, Left = -2.392, Right = -2.233 }
BeamInfo { BeamNumber = 34, Center = -2.154, Left = -2.233, Right = -2.075 }
BeamInfo { BeamNumber = 35, Center = -1.995, Left = -2.075, Right = -1.916 }
BeamInfo { BeamNumber = 36, Center = -1.836, Left = -1.916, Right = -1.757 }
BeamInfo { BeamNumber = 37, Center = -1.677, Left = -1.757, Right = -1.598 }
BeamInfo { BeamNumber = 38, Center = -1.518, Left = -1.598, Right = -1.438 }
BeamInfo { BeamNumber = 39, Center = -1.359, Left = -1.438, Right = -1.279 }
BeamInfo { BeamNumber = 40, Center = -1.199, Left = -1.279, Right = -1.119 }
BeamInfo { BeamNumber = 41, Center = -1.039, Left = -1.119, Right = -0.96 }
BeamInfo { BeamNumber = 42, Center = -0.88, Left = -0.96, Right = -0.8 }
BeamInfo { BeamNumber = 43, Center = -0.72, Left = -0.8, Right = -0.64 }
BeamInfo { BeamNumber = 44, Center = -0.56, Left = -0.64, Right = -0.48 }
BeamInfo { BeamNumber = 45, Center = -0.4, Left = -0.48, Right = -0.32 }
BeamInfo { BeamNumber = 46, Center = -0.24, Left = -0.32, Right = -0.16 }
BeamInfo { BeamNumber = 47, Center = -0.08, Left = -0.16, Right = 0 }
BeamInfo { BeamNumber = 48, Center = 0.08, Left = 0, Right = 0.16 }
BeamInfo { BeamNumber = 49, Center = 0.24, Left = 0.16, Right = 0.32 }
BeamInfo { BeamNumber = 50, Center = 0.4, Left = 0.32, Right = 0.48 }
BeamInfo { BeamNumber = 51, Center = 0.56, Left = 0.48, Right = 0.64 }
BeamInfo { BeamNumber = 52, Center = 0.72, Left = 0.64, Right = 0.8 }
BeamInfo { BeamNumber = 53, Center = 0.88, Left = 0.8, Right = 0.96 }
BeamInfo { BeamNumber = 54, Center = 1.039, Left = 0.96, Right = 1.119 }
BeamInfo { BeamNumber = 55, Center = 1.199, Left = 1.119, Right = 1.279 }
BeamInfo { BeamNumber = 56, Center = 1.359, Left = 1.279, Right = 1.438 }
BeamInfo { BeamNumber = 57, Center = 1.518, Left = 1.438, Right = 1.598 }
BeamInfo { BeamNumber = 58, Center = 1.677, Left = 1.598, Right = 1.757 }
BeamInfo { BeamNumber = 59, Center = 1.836, Left = 1.757, Right = 1.916 }
BeamInfo { BeamNumber = 60, Center = 1.995, Left = 1.916, Right = 2.075 }
BeamInfo { BeamNumber = 61, Center = 2.154, Left = 2.075, Right = 2.233 }
BeamInfo { BeamNumber = 62, Center = 2.313, Left = 2.233, Right = 2.392 }
BeamInfo { BeamNumber = 63, Center = 2.471, Left = 2.392, Right = 2.55 }
BeamInfo { BeamNumber = 64, Center = 2.63, Left = 2.55, Right = 2.709 }
BeamInfo { BeamNumber = 65, Center = 2.788, Left = 2.709, Right = 2.868 }
BeamInfo { BeamNumber = 66, Center = 2.947, Left = 2.868, Right = 3.026 }
BeamInfo { BeamNumber = 67, Center = 3.105, Left = 3.026, Right = 3.185 }
BeamInfo { BeamNumber = 68, Center = 3.264, Left = 3.185, Right = 3.343 }
BeamInfo { BeamNumber = 69, Center = 3.422, Left = 3.343, Right = 3.501 }
BeamInfo { BeamNumber = 70, Center = 3.581, Left = 3.501, Right = 3.66 }
BeamInfo { BeamNumber = 71, Center = 3.739, Left = 3.66, Right = 3.818 }
BeamInfo { BeamNumber = 72, Center = 3.897, Left = 3.818, Right = 3.976 }
BeamInfo { BeamNumber = 73, Center = 4.055, Left = 3.976, Right = 4.134 }
BeamInfo { BeamNumber = 74, Center = 4.213, Left = 4.134, Right = 4.291 }
BeamInfo { BeamNumber = 75, Center = 4.37, Left = 4.291, Right = 4.448 }
BeamInfo { BeamNumber = 76, Center = 4.526, Left = 4.448, Right = 4.605 }
BeamInfo { BeamNumber = 77, Center = 4.682, Left = 4.605, Right = 4.76 }
BeamInfo { BeamNumber = 78, Center = 4.838, Left = 4.76, Right = 4.916 }
BeamInfo { BeamNumber = 79, Center = 4.993, Left = 4.916, Right = 5.07 }
BeamInfo { BeamNumber = 80, Center = 5.148, Left = 5.07, Right = 5.225 }
BeamInfo { BeamNumber = 81, Center = 5.302, Left = 5.225, Right = 5.379 }
BeamInfo { BeamNumber = 82, Center = 5.456, Left = 5.379, Right = 5.533 }
BeamInfo { BeamNumber = 83, Center = 5.609, Left = 5.533, Right = 5.686 }
BeamInfo { BeamNumber = 84, Center = 5.763, Left = 5.686, Right = 5.839 }
BeamInfo { BeamNumber = 85, Center = 5.915, Left = 5.839, Right = 5.992 }
BeamInfo { BeamNumber = 86, Center = 6.068, Left = 5.992, Right = 6.144 }
BeamInfo { BeamNumber = 87, Center = 6.22, Left = 6.144, Right = 6.296 }
BeamInfo { BeamNumber = 88, Center = 6.372, Left = 6.296, Right = 6.448 }
BeamInfo { BeamNumber = 89, Center = 6.523, Left = 6.448, Right = 6.599 }
BeamInfo { BeamNumber = 90, Center = 6.674, Left = 6.599, Right = 6.749 }
BeamInfo { BeamNumber = 91, Center = 6.824, Left = 6.749, Right = 6.899 }
BeamInfo { BeamNumber = 92, Center = 6.974, Left = 6.899, Right = 7.049 }
BeamInfo { BeamNumber = 93, Center = 7.123, Left = 7.049, Right = 7.199 }
BeamInfo { BeamNumber = 94, Center = 7.271, Left = 7.199, Right = 7.345 }
BeamInfo { BeamNumber = 95, Center = 7.419, Left = 7.345, Right = 7.493 }
